### PR TITLE
#122: Use express validator

### LINKED ISSUE #122: Use the express-validator module
Use [Express Validator](https://express-validator.github.io/docs/) to do data validation before sending the data to the database.

### PR DESCRIPTION
**Pull request by thomastai1666**
Closes #122, Uses the express-validator module prior to sending any data originating from the request to the database.

### CODE REVIEW

**Review by Jen-Lopez — APPROVED** [review]

File: back-end/routes/cuisine.js [hunk 1]
```diff
@@ -1,47 +1,66 @@
...
+      if (err) {
+        console.log("Something wrong when finding the group");
+        res.status(500);
+        res.send(err);
+        return;
+      }
 
-       if(count){ //if count is defined
+      const count = doc.selectedCuisines.find((selectedCuisine) => {
+        return cuisine == selectedCuisine.cuisine;
```
Comment (yunko1803) [inline back-end/routes/cuisine.js:28]: ===

File: back-end/routes/documenu.js [hunk 2]
```diff
@@ -1,47 +1,64 @@
...
-    })
-});
+      }
+
+      const location = doc.location;
+      const params = {
+        lat: location.latitude,
+        lon: location.longitude,
+        distance: 5,
+        search: searchKeyWord,
```
Comment (yunko1803) [inline back-end/routes/documenu.js:36]: what is this search for?

File: back-end/routes/room.js [hunk 3]
```diff
@@ -1,41 +1,66 @@
...
+    return res.status(400).json({ errors: errors.array() });
+  }
+  const invCode = req.query.inviteCode;
+  Group.find({ groupId: invCode }, (err, result) => {
+    if (err) {
+      console.log(err);
+      res.send({ valid: false, msg: "An error occured." });
+      return;
+    }
+    // console.log(result);
```
Comment (yunko1803) [inline back-end/routes/room.js:55]: remove

File: back-end/routes/user.js [hunk 4]
```diff
@@ -15,19 +25,25 @@ router.post("/user", function (req, res) {
         res.send({ success: false });
         return;
       }
-    //   console.log(result);
-      Group.findOneAndUpdate({groupId: id}, {$addToSet: {friends: newUser}}, {new: true}, (err, doc) => {
-        if (err) {
+      //   console.log(result);
```
Comment (yunko1803) [inline back-end/routes/user.js:28]: remove

File: back-end/routes/user.js [hunk 5]
```diff
@@ -15,19 +25,25 @@ router.post("/user", function (req, res) {
...
+        { new: true },
+        (err, doc) => {
+          if (err) {
             // console.log("Something wrong when updating the data");
             res.status(500);
             res.send({ success: false });
-            return
+            return;
+          }
+          // console.log(doc);
```
Comment (yunko1803) [inline back-end/routes/user.js:40]: remove comment

**Review by yunko1803 — COMMENTED** [review]
Nicely done!

just some slight issues

File: back-end/routes/cuisine.js (hunk 1 above)
Comment (thomastai1666) [reply back-end/routes/cuisine.js:28]:
```suggestion
        return cuisine === selectedCuisine.cuisine;
```

File: back-end/routes/room.js (hunk 3 above)
Comment (thomastai1666) [reply back-end/routes/room.js:55]:
```suggestion
```

File: back-end/routes/user.js (hunk 4 above)
Comment (thomastai1666) [reply back-end/routes/user.js:28]:
```suggestion
```

File: back-end/routes/user.js (hunk 5 above)
Comment (thomastai1666) [reply back-end/routes/user.js:40]:
```suggestion
```

File: back-end/routes/documenu.js (hunk 2 above)
Comment (thomastai1666) [reply back-end/routes/documenu.js:36]: I think search is the keyword we use when we filter through dishes on the preferred dish page

**Review by Jen-Lopez — APPROVED** [review]

**Review by yunko1803 — APPROVED** [review]
LGTM

**Pull request merged**